src/runtime/metal/metal_module.mm

```diff
@@ -43,7 +44,9 @@
  public:
   explicit MetalModuleNode(std::string data, std::string fmt,
                            std::unordered_map<std::string, FunctionInfo> fmap, std::string source)
-      : data_(data), fmt_(fmt), fmap_(fmap), source_(source) {}
+      : data_(data), fmt_(fmt), fmap_(fmap), source_(source) {
+    parsed_kernels_ = SplitKernels(GetSource(fmt_));
```

Comment:
How about passing `data` directly here to `SplitKernels`

Reply:
You are right. Thank you. We can use `data_` here.